src/Pos/Update/Poll/Logic/Apply.hs

```diff
@@ -283,16 +283,22 @@ verifyBlockVersion upId UpdateProposal {..} = do
 -- Here we check that proposal has at least 'genesisUpdateProposalThd'
 -- stake of total stake in all positive votes for it.
 verifyProposalStake
-    :: (MonadError PollVerFailure m)
+    :: (MonadPollRead m, MonadError PollVerFailure m)
     => Coin -> [(UpdateVote, Coin)] -> UpId -> m ()
 verifyProposalStake totalStake votesAndStakes upId = do
-    let threshold = applyCoinPortion genesisUpdateProposalThd totalStake
+    let threshold =
+          coinToInteger $ applyCoinPortion genesisUpdateProposalThd totalStake
     let votesSum =
             sumCoins . map snd . filter (uvDecision . fst) $ votesAndStakes
-    when (coinToInteger totalStake < votesSum) $
```

Comment:
Yes, this line was completely stupid. It worked because there were two bugs: wrong argument (total stake instead of threshold) *and* wrong operation (`<` instead of `>`).